test/mpi/test_mpi.c

```diff
@@ -81,6 +85,13 @@ int ucc_mpi_test_init(int argc, char **argv, uint64_t coll_types,
     MPI_Comm_rank(MPI_COMM_WORLD, &rank);
     MPI_Comm_size(MPI_COMM_WORLD, &size);
 
+#ifdef UCC_TEST_WITH_CUDA
```

Comment:
the test_mpi is completely reworked in kn_allreduce PR, will need to add this change there. just to keep in mind.